src/igr-tests/igr_functions

```diff
@@ -0,0 +1,239 @@
+            export DINITMONITOR="$EXEC_PREFIX/dinit-monitor"
+        else
+            return 1
+        fi
+    fi
+    return 0
+}
+
+## Change directory to run-test.sh location
+cd "$(dirname "$0")" || error "Can't change directory to script location."
+
+## Prepare $IGR_OUTPUT
+TEST_NAME="${PWD##*/}"
+if [ -z "$TEST_NAME" ]; then error "Failed to guess test name."; fi
+if [ -z "${IGR_OUTPUT:-}" ]; then
+    IGR_OUTPUT="$PWD"
+    if [ -z "$IGR_OUTPUT" ]; then error "Failed to guess igr output files location."; fi
+    export IGR_OUTPUT
+else
+    # Igr tests are used by meson.
+    mkdir -p "$IGR_OUTPUT/$TEST_NAME/"
+    IGR_OUTPUT="$IGR_OUTPUT/$TEST_NAME/"
+    export IGR_OUTPUT
+fi
+mkdir -p "$IGR_OUTPUT"/output/
+export SOCKET="$IGR_OUTPUT/socket"
+if [ -n "${DEBUG:-}" ]; then
+    QUIET=""
+else
+    QUIET="--quiet"
+fi
+
+## Integration tests helper functions
+# This function spawn a dinit daemon with temp socket.
+# It's pass its flags as dinit daemon flags.
+# RESULT: return 0 & sets "$SOCKET" & "$DINITPID" variables on success.
+#         throw an error on failure.
+spawn_dinit() {
+    find_dinit || error "Cannot find dinit exec path." "Check 'EXEC_PREFIX' variable value"
+    "$DINIT" $QUIET -u -d sd -p "$SOCKET" "$@" &
+    DINITPID=$!
+    # Wait for Dinit socket shows up.
+    TIMEOUT=0
+    while [ ! -e "$SOCKET" ]; do
+        if [ $TIMEOUT -le 600 ]; then
+            sleep 0.1
+            TIMEOUT=$((TIMEOUT+=1))
+        else
+            error "Dinit reaches timeout but socket didn't create."
+        fi
+    done
+    return 0
+}
+
+# This function stop current Dinit instance.
+# Doesn't accept anything.
+# RESULT: return 0 if Dinit stops.
+#         return 0 and throw a warning if Dinit already stopped.
+#         throw an error on failure.
+# NOTE: Don't use error(), using error() will result a inifinite cycle.
+stop_dinit() {
+    if [ ! -e "$SOCKET" ]; then
+        warning "stop_dinit() called but cannot find any running dinit instance!"
+        return 0
+    fi
+    if find_dinitctl && $DINITCTL $QUIET shutdown -p "$SOCKET"; then
+        wait "$DINITPID"
+        return 0
+    else
+        warning "Cannot stopping dinit via dinitctl." "Fallback to killing DINITPID."
+        kill "$DINITPID" || ( echo "${TEST_NAME:-}: Cannot stop Dinit instance!" && exit 1 )
+        wait "$DINITPID"
+    fi
+}
+
+# This function spawn a dinit daemon but blocks until dinit exits.
+# It's pass its flags as dinit daemon flags.
+# RESULT: waits for dinit daemon and return exit code of dinit on sccuess.
+#         throw an error on failure.
+spawn_dinit_oneshot() {
+    find_dinit || error "Cannot find dinit exec path." "Check 'EXEC_PREFIX' variable value."
+    "$DINIT" $QUIET -u -d sd -p "$SOCKET" "$@"
+}
+
+# This function find dinitctl and allow access to dinit daemon via dinitctl.
+# It's pass its flags as dinitctl flags.
+# RESULT: Any return code from dinitctl on sccuess.
+#         Throw an error on failure.
+dinitctl() {
```

Comment:
Please rename this - it's very confusing that tests run "dinitctl" but they are actually calling this function instead of running the real `dinitctl`

Reply:
Maybe `run_dinitctl`